gaj23/get-going-gone
# Gaj/feature/dashboard build

#### Purpose: 
> Check all that apply.

### What does this PR do? 
> Clearly and concisely describe changes made. 

This PR accomplishes the following: 
 - main dashboard structure 
 - set up for dom manipulation in HTML
 - succeeds in a lighthouse evaluation of 100% 
 - successful WAVE evaluation
 - functional dark mode toggle 
 - set up API calls 

Additional file structure changes were made for clarification of scss styling. 

### Problems Encountered & Resulting Solutions  
> State any problems encountered 
> Describe your code changes in detail for reviewers.

Problem solving when building out the dark mode toggle proved constant awareness to contrasting colors and dom updates of small pieces (such as button backgrounds as well as font changes) so that in both modes, information is easily conveyed. 

I encountered difficulties applying Promise.all() and knowing when I was receiving data. Help was found with my cohort mate and accountabilabuddy, Alice Ruppert (@srslie). 

### Where should the reviewer start?
> What file(s) and line(s) are where changes are made?

- src/css
  - _darkMode.scss for darkMode styling

- src/scripts.js 
  - toggleDarkMode lines 28 - 43
  - API calls lines 20 - 25 (see also apiCalls.js) 

### How should this be tested?
> What steps need to be taken to confirm functionality? 

Please confirm toggle functionality by loading the deployed sites.
Please confirm API success by calling upon the globally declared variables of `travelers`, `trips`, and `destinations`. 

### Steps To Take
> Any questions and possible steps moving forward.

Because toggleDarkMode has pieces that also manage dom manipulation, some pieces of that should be moved in the domUpdates files instead. 

#### Checklist:
